app/waf/src/repositories/common.ts

```diff
@@ -0,0 +1,28 @@
+import {Entity} from '@loopback/repository';
+import {DataObject, Options, DefaultCrudRepository} from '@loopback/repository';
+import uuid = require('uuid');
+
+export class CommonRepository<
+  T extends Entity,
+  ID
+> extends DefaultCrudRepository<T, ID> {
+  create(data: DataObject<T>, options?: Options): Promise<T> {
```

Comment:
Since it's a creation function, I think there's no need to check `id` and `createdAt`. 
How about generating them directly?

Reply:
The API will have validator to allow/deny those input. But is there any possibility that we would need to specify those properties in other layer, like controller? I think maybe or maybe not. What do you think?

Reply:
Just thinking that if there's 'other layer' to specify those properties, we may leverage this repository to do that, not directly from 'other layer'.

Reply:
OK